### PR TITLE
[5.7] Use PHP's configured default character set when outputting from "e" helper function

### PR DESCRIPTION
**Pull request by maaarghk**
Rationale is in #25689

### CODE REVIEW

**Comment by mfn** [comment]
FTR the default for `default_charset` _is_ `UTF-8` (as per php.ini distributed by php.net; no idea about distributions and their choiced).

I just mentioned it because I didn't see it explicitly mentioned in the referenced issue and maybe it's not clear to everyone.

So on the outset, this change looks reasonable and would only cause changes for people having a different `default_charset` at the moment which however would be a weird thing because[ other functions as affected as well as we all know](http://php.net/manual/en/ini.core.php#ini.default-charset).

TL;DR: the patch and rational sounds reasonable to me.

**Comment by maaarghk** [comment]
Hi, thanks for reading and I agree. In the issue, that is what I tried to put across in the sentence: "...Laravel assumes UTF-8 is the default_charset because that is the configuration on the minimum required version of PHP." You're right that it isn't too clear though plus it _is_ buried in a wall of text!

**Comment by taylorotwell** [comment]
Potential breaking change.

**Comment by maaarghk** [comment]
@taylorotwell I was fairly sure that'd be the answer :)

Do you think either (a) there is somewhere else this could be merged to or (b) there is some way to prove with tests that it is not a breaking change?

**Comment by browner12** [comment]
you can send the PR to the master branch, which will go into our next major release (5.8).

I think by "potential breaking change" he means it has the potential to affect users, so it **is a breaking change**.
